### PR TITLE
Bug 1318709 - Create pre-release Longitudinal

### PR DESCRIPTION
**Pull request by fbertsch**
This is now per-process and each opt-in or opt-out, not both.

### CODE REVIEW

File: src/main/scala/com/mozilla/telemetry/histograms/Histogram.scala [hunk 1]
```diff
@@ -120,7 +126,7 @@ object Histograms {
 
       val pretty = for {
         (k, v) <- result
-        if v.getOrElse("releaseChannelCollection", Some("opt-in")) == Some("opt-out")
+        if (includeOptin && v.getOrElse("releaseChannelCollection", Some("opt-in")) == Some("opt-in")) || ((!includeOptin) && v.getOrElse("releaseChannelCollection", Some("opt-in")) == Some("opt-out"))
```
Comment (harterrt) [inline src/main/scala/com/mozilla/telemetry/histograms/Histogram.scala]: Let's refactor this logic to a separate function or simplify. 

It looks like this is splitting the longitudinal sets into two distinct sets of opt-in or non-opt-in (as discussed in the bug). If that's the case then `includeOptin` is misleading me. When I hear `includeOptin` I expect it to augment the set, not split. 

I'm having some difficulty typing `Optin`, it feels like a misspelling of `Option`. Does it make sense to use `OptIn` instead?

File: src/main/scala/com/mozilla/telemetry/histograms/Histogram.scala [hunk 2]
```diff
@@ -129,10 +135,19 @@ object Histograms {
         val high = v.getOrElse("high", None).asInstanceOf[Option[Int]]
         val nBuckets = v.getOrElse("n_buckets", None).asInstanceOf[Option[Int]]
 
-        def addSuffixes(key: String, histogram: HistogramDefinition): List[(String, HistogramDefinition)] = {
-          suffixes.map(suffix => (key + suffix, histogram))
+        def addProcessSuffixes(processType: Option[String])(key: String, histogram: HistogramDefinition): List[(String, HistogramDefinition)] = {
```
Comment (harterrt) [inline src/main/scala/com/mozilla/telemetry/histograms/Histogram.scala]: Can we remove this intermediate function by changing the way we create `suffixes`? See the comment above where we define suffixes.

File: src/main/scala/com/mozilla/telemetry/scalars/Scalars.scala [hunk 3]
```diff
@@ -42,6 +45,8 @@ package object Scalars {
       // Emit a scalar definition for each scalar.
       val scalarDefinitions = for {
         (scalarName, v) <- flattenedScalars
+        if (optinOnly && v.asInstanceOf[util.LinkedHashMap[String, Any]].getOrElse("release_channel_collection", "opt-in") == "opt-in") ||
```
Comment (harterrt) [inline src/main/scala/com/mozilla/telemetry/scalars/Scalars.scala]: Same as above, let's refactor this logic to be easier to read.

**Comment by fbertsch** [comment]
Removing per-process stuff, so going to file a new PR.

**Pull request closed without merging**